tools/fv_nudge.F90

```diff
@@ -3545,16 +3545,17 @@ subroutine pmaxmin( qname, a, imax, jmax, fac )
       class(*) a(imax,jmax)
       class(*) fac                     ! multiplication factor
 
-      real(r4_kind) qmin(jmax), qmax(jmax)
+      real(r4_kind), dimension(:), allocatable :: qmin, qmax
```

Comment:
Why do these need to be allocatable arrays when we already know their length?

Reply:
`allocatable` is used to save memory: Depending on whether it is dealing with single or double data (from FMS), either (`qmin`, `qmax`) OR (`qmin8`, `qmax8`) is required, but not both; the selection and allocation is done inside the `select type` construct below.